lib/conjur/cli.rb

```diff
@@ -121,19 +121,26 @@ def init!
     
     on_error do |exception|
       require 'rest-client'
-      if exception.is_a?(RestClient::Exception)
-        begin
-          body = JSON.parse(exception.response.body)
-          $stderr.puts body['error']
-        rescue
-          $stderr.puts exception.response.body if exception.response
+      require 'patches/conjur/error'
+
+      use_default_handler = true
+
+      if exception.is_a?(RestClient::Exception) && exception.response
+        err = Conjur::Error.create exception.response.body
+        if err
+          exception = err
+          $stderr.puts "error: " + err.message
+          use_default_handler = false
+        else
+          $stderr.puts exception.response.body
         end
       end
       require 'conjur/log'
       if Conjur.log
         Conjur.log << "error: #{exception}\n#{exception.backtrace.join("\n") rescue 'NO BACKTRACE?'}"
```

Comment:
In other words, I think this flow could use some comments about what's happening and why :-)